src/core/media/player_skeleton.cpp

```diff
@@ -347,9 +353,31 @@ struct media::PlayerSkeleton::Private
                 remote_error->emit(e);
             });
 
-            buffering_changed.connect([remote_buffering_changed](int value)
+            buffering_changed.connect([this, remote_buffering_changed](int value)
             {
-                remote_buffering_changed->emit(value);
+                auto now = std::chrono::steady_clock::now();
+                auto time_since_last_emission = now - buffering_signal_last_emission;
+                if (value >= 100 ||
+                    time_since_last_emission > std::chrono::milliseconds(500)) {
+                    // Send the signal immediately
+                    buffering_signal_timeout.cancel();
+                    buffering_signal_value = -1;
+                    remote_buffering_changed->emit(value);
+                    buffering_signal_last_emission = now;
+                } else {
+                    if (buffering_signal_value < 0) {
```

Comment:
The buffering will not go below 0, right?

Reply:
Correct. But I use `-1` as a sentinel value to indicate that there is no queued signal emission.